File: algorandsamples/mypythondemo/Indexer/SearchTransactionsNote.py
import json
# requires Python SDK version 1.3 or higher
from algosdk.v2client import indexer

data = {
    "indexer_token": "",
    "indexer_address": "http://localhost:8980"
}
# instantiate indexer client
myindexer = indexer.IndexerClient(**data)
import base64
encodednote = base64.b64encode('showing prefix'.encode())
data = {
    "note_prefix": base64.b64decode(encodednote)
}
response = myindexer.search_transactions(**data)
print("note_prefix = " +
      json.dumps(response, indent=2, sort_keys=True))

File: algorandsamples/v2/mygodemo/src/indexer/SearchTransactionsNote.go
package main

import (
	"context"
	"encoding/json"
	"fmt"

	"github.com/algorand/go-algorand-sdk/client/v2/indexer"
)

// indexer host
const indexerAddress = "http://localhost:8981"
const indexerToken = ""

// query parameters
var minAmount uint64 = 10

// var data = "showing prefix"
// var encodedNote = base64.StdEncoding.EncodeToString([]byte(data))
var notePrefix = "Russ"
var round uint64 = 11563027
func main() {

	// Create an indexer client
	indexerClient, err := indexer.MakeClient(indexerAddress, indexerToken)
	if err != nil {
		return
	}

	// Query
	result, err := indexerClient.SearchForTransactions().MinRound(round).NotePrefix([]byte(notePrefix)).Do(context.Background())

	// Print the results
	JSON, err := json.MarshalIndent(string(result.Transactions[0].Note), "", "\t")
	fmt.Printf(string(JSON) + "\n")
}
